### ISSUE TITLE
julia-debug does not build on Power

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#17380]
We are already using LLVM 3.8 on power. @vtjnash says this will be fixed with llvm 3.8.1.

````
julia-debug: /home/vinchhid/julia/src/gc.c:792: jl_gc_pool_alloc: Assertion `ptls->gc_state == 0' failed.

signal (6): Aborted
while loading sysimg.jl, in expression starting on line 3
jl_gc_pool_alloc at /home/vinchhid/julia/src/gc.c:792
jl_gc_alloc_ at /home/vinchhid/julia/src/julia_internal.h:134
jl_gc_alloc at /home/vinchhid/julia/src/gc.c:1848
_new_array_ at /home/vinchhid/julia/src/array.c:95
_new_array at /home/vinchhid/julia/src/array.c:149
jl_alloc_array_1d at /home/vinchhid/julia/src/array.c:346
jl_alloc_vec_any at /home/vinchhid/julia/src/array.c:396
jl_exprn at /home/vinchhid/julia/src/alloc.c:1311
scm_to_julia_ at /home/vinchhid/julia/src/ast.c:485
scm_to_julia at /home/vinchhid/julia/src/ast.c:337
jl_parse_eval_all at /home/vinchhid/julia/src/ast.c:695
jl_load at /home/vinchhid/julia/src/toplevel.c:596
exec_program at /home/vinchhid/julia/ui/repl.c:66
true_main at /home/vinchhid/julia/ui/repl.c:119
main at /home/vinchhid/julia/ui/repl.c:232
Allocations: 102413 (Pool: 102343; Big: 70); GC: 0
````

**Comment by yuyichao** [comment]
The failure is very weird. We shouldn't be doing the GC transition at many places yet and it haven't even does a GC. @vtjnash Are you suggesting that LLVM write to the wrong address and set the `gc_state` incorrectly somehow?

**Comment by ViralBShah** [comment]
I have verified that this does indeed work with llvm 3.8.1, which is what we are now using for Julia 0.5.

**Comment by ViralBShah** [comment]
Fixed by 10b67fcc25e7e8960813bee236fdc8d830b8d9e4
